pkg/controller/labelidentity/label_group_index.go

```diff
@@ -0,0 +1,514 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package labelidentity
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/controller/types"
+	antreatypes "antrea.io/antrea/pkg/controller/types"
+)
+
+const (
+	// Cluster scoped selectors are stored under empty Namespace in indice.
+	emptyNamespace = ""
+	policyIndex    = "policyIndex"
+)
+
+var (
+	// eventChanSize is declared as a variable to allow overriding for testing.
+	eventChanSize = 1000
+)
+
+// eventHandler is the registered callback for policy re-sync
+type eventHandler func(policyKey string)
+
+type Interface interface {
+	// AddSelector adds or updates a selectorItem when a new selector is added to a policy.
+	AddSelector(selector *types.GroupSelector, policyKey string) []uint32
+	// DeleteSelector deletes or updates a selectorItem when a selector is deleted from a policy.
+	DeleteSelector(selectorKey string, policyKey string)
+	// GetLabelIdentityIDs retrieves the label identity IDs selected by the provided selectorItem keys.
+	GetLabelIdentityIDs(selectorKey string) []uint32
+	// SetPolicySelectors registers a policy's selectors with the index.
+	SetPolicySelectors(selectors []*antreatypes.GroupSelector, policyKey string) []uint32
+	// DeletePolicySelectors removes any selectors from referring to the policy being deleted.
+	DeletePolicySelectors(policyKey string)
+	// AddLabelIdentity adds LabelIdentity-ID mapping to the index.
+	AddLabelIdentity(labelKey string, id uint32)
+	// DeleteLabelIdentity deletes a LabelIdentity from the index.
+	DeleteLabelIdentity(labelKey string)
+	// AddEventHandler registers an eventHandler with the index.
+	AddEventHandler(handler eventHandler)
+	// Run starts the index.
+	Run(stopCh <-chan struct{})
+	// HasSynced returns true if the interface has been initialized with the full lists of LabelIdentities.
+	HasSynced() bool
+}
+
+type selectorItemUpdate string
```

Comment:
Maybe name it as `selectorItemUpdateEvent` or `selectorItemUpdateTrigger`?

Reply:
Done